cedar-policy-core/src/error_macros.rs

```diff
@@ -21,13 +21,14 @@
 macro_rules! impl_diagnostic_from_source_loc_field {
     ( $i:ident ) => {
         fn source_code(&self) -> Option<&dyn miette::SourceCode> {
-            Some(&self.$i.src as &dyn miette::SourceCode)
+            self.$i.as_ref().map(|l| &l.src as &dyn miette::SourceCode)
```

Comment:
shouldn't we just use the `impl_diagnostic_from_source_loc_opt_field` macro (below) instead of changing this one?

If we do change this one, let's update the comment to indicate `$i` should be type `Option<Loc>`

Reply:
Yeah. Is it used in any other crates we control?

Reply:
I can't find them anywhere in this repo nor in cedar-spec, so I'm removing it. Also, I was doing the same thing with the other macro... rewriting the non-optional. So I'm removing both non-optional, and using the optional versions only.